gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1693,6 +1646,34 @@ protected GenerateDataTemplateTask configureDataTemplateGeneration(Project proje
     prepareLegacySchemasForPublishTask.dependsOn(destroyStaleFiles);
     dataTemplateJarDepends.add(prepareLegacySchemasForPublishTask);
 
+    // extension schema directory
+    File extensionSchemaDir = project.file(getExtensionSchemaPath(project, sourceSet));
+
+    if (!SharedFileUtils.getSuffixedFiles(project, extensionSchemaDir, PDL_FILE_SUFFIX).isEmpty()) {
+      // Validate extension schemas if extension schemas are provided.
+      ValidateExtensionSchemaTask validateExtensionSchemaTask = project.getTasks()
+          .create(sourceSet.getTaskName("validate", "ExtensionSchema"), ValidateExtensionSchemaTask.class, task -> {
+            task.setInputDir(extensionSchemaDir);
+            task.setResolverPath(
+                getDataModelConfig(project, sourceSet).plus(project.files(getDataSchemaPath(project, sourceSet))));
+            task.setClassPath(project.getConfigurations().getByName("pegasusPlugin"));
+            if (isPropertyTrue(project, ENABLE_ARG_FILE)) {
+              task.setEnableArgFile(true);
+            }
+          });
+
+      Task prepareExtensionSchemasForPublishTask = project.getTasks()
+          .create(sourceSet.getName() + "CopyExtensionSchemas", Sync.class, task ->
+          {
+            task.from(extensionSchemaDir, syncSpec -> syncSpec.include("**/*" + PDL_FILE_SUFFIX));
```

Comment:
So we're only supporting extensions as PDLs from the get-go? Good 😃

Reply:
Yes, this feature is only supported for PDL.